File: client-app/src/app/models/purchaseOrder.ts
import { Company } from "./company";
import { Product } from "./product";

export interface PurchaseOrder {
    id: string,
    productId :string,
    product? :Product | null,
    companyId : string,
    company?:Company | null,
    quantity: number,
    contractDate : Date | null,
    exportDate : Date | null,
    exportCountry : string,
    isDelivered : boolean
}

File: Domain/PurchaseOrder.cs

namespace Domain
{
    public class PurchaseOrder
    {
        public Guid Id { get; set; }
        public Guid ProductId { get; set; }
        public Product Product { get; set; }
        public Guid CompanyId { get; set; }
        public Company Company { get; set; }
        public long Quantity { get; set; }
        public DateTime ContractDate { get; set; }
        public DateTime ExportDate { get; set; }
        public string ExportCountry { get; set; }
    }
}